examples/simple_with_session.rs

```diff
@@ -77,8 +75,14 @@ async fn authenticated_index(user: &mut GooseUser) -> GooseTaskResult {
     // This will panic if the session is missing or if the session is not of the right type
     // use `get_session_data` to handle missing session
     let session = user.get_session_data_unchecked::<Session>();
-    let request = user.goose_get("/")?.bearer_auth(&session.jwt_token);
-    let _goose = user.goose_send(request, None).await?;
+    let goose_request = GooseRequest::builder()
+        .request_builder(
+            user.get_request_builder(&GooseMethod::Get, "/")?
+                .bearer_auth(&session.jwt_token),
```

Comment:
I would just use a helper var, so that it looks like this:

```
let reqwest_request_builder =  user.get_request_builder(&GooseMethod::Get, "/")?
                .bearer_auth(&session.jwt_token),

let goose_request = GooseRequest::builder()
        .set_request_builder(reqwest_request_builder)
        .build();
```

However for this common example I would just add bearer_auth as a function to the GooseRequest builder and pass it through to Reqwest.

Easy.

Reply:
As both are built at different times for maximum flexibility, there's no clear way to accept it in the `GooseRequestBuilder` and pass it to the `RequestBuilder` -- in some workflows that could work, but not in all.

Reply:
I've split it out as you suggested, which I agree makes it easier to understand.